code/ui/alivehud/c4/C4ArmControl.cs

```diff
@@ -0,0 +1,214 @@
+            _content.UpdateTimer(timerString);
+
+            int wires = C4Entity.TimerPresets[(int) preset].wires;
+            int defuseChance = (1f / wires * 100f).FloorToInt();
+
+            _content.UpdatePresetInfo($"{defuseChance}% chance to defuse");
+        }
+
+        private class Header : TTTPanel
+        {
+            private Label _title;
+            private Button _closeButton;
+
+            public Header(Panel parent)
+            {
+                Parent = parent;
+
+                _title = Add.Label("C4 Control", "title");
+                _closeButton = Add.Button("Close", () => { ((C4ArmControl)Parent).Close(); });
+            }
+        }
+
+        private class Content : TTTPanel
+        {
+            private Display _display;
+
+            public Content(Panel parent)
+            {
+                Parent = parent;
+
+                _display = new(this);
+            }
+
+            public void UpdateTimer(string timerString)
+            {
+                _display.UpdateTimer(timerString);
+            }
+
+            public void UpdatePresetInfo(string infoString)
+            {
+                _display.UpdatePresetInfo(infoString);
+            }
+
+            public class Display : TTTPanel
+            {
+                private Label _timer;
+                private Presets _presets;
+
+                public Display(Panel parent)
+                {
+                    Parent = parent;
+
+                    _timer = Add.Label("00:00", "timer");
+                    _presets = new(this);
+                }
+
+                public void UpdateTimer(string timerString)
+                {
+                    _timer.Text = timerString;
+                }
+
+                public void UpdatePresetInfo(string infoString)
+                {
+                    _presets.UpdatePresetInfo(infoString);
+                }
+
+                public class Presets : TTTPanel
+                {
+                    private PresetButtons _presetButtons;
+                    private Label _presetInfo;
+
+                    public Presets(Panel parent)
+                    {
+                        Parent = parent;
+
+                        _presetInfo = Add.Label("Select a timer preset:");
+                        _presetButtons = new(this);
+                    }
+
+                    public void UpdatePresetInfo(string infoString)
+                    {
+                        _presetInfo.Text = infoString;
+                    }
+
+                    public class PresetButtons : TTTPanel
+                    {
+                        private Button _presetShort;
+                        private Button _presetMedium;
+                        private Button _presetLong;
+
+                        public PresetButtons(Panel parent)
+                        {
+                            Parent = parent;
+
+                            var controls = (C4ArmControl) Parent.Parent.Parent.Parent;
```

Comment:
Pain.Pain.Pain.Pain;

Reply:
New UI implementation will take care of this problem

Reply:
Anyways, pls add the `C4ArmControl` as a var

Reply:
Oh yea, and `var` ^^